examples/cn0548_py.py

```diff
@@ -0,0 +1,176 @@
+    import tkinter.scrolledtext as tkscrolled
+
+except:
+    print("Please install tkinter")
+try:
+    import csv
+except:
+    print("Please install csv")
+try:
+    import pandas as pd
+except:
+    print("Please install pandas")
+try:
+    import numpy as np
+except:
+    print("Please install numpy")
+
+try:
+    import matplotlib.pyplot as plt
+    from matplotlib.backends.backend_tkagg import (
+        FigureCanvasTkAgg,
+        NavigationToolbar2Tk,
+    )
+except:
+    print("Please install matplotlib")
+
+# Setup device - get device data
+def read_device():
+
+    global ad7799, gain, voltage0, voltage1, voltage2
+
+    gain_val = ad7799.gain
+    voltage0_val = ad7799.channel[0].value
+    voltage1_val = ad7799.channel[1].value
+    voltage2_val = ad7799.channel[2].value
+
+    voltage0.set(voltage0_val)
+    voltage1.set(voltage1_val)
+    voltage2.set(voltage2_val)
+    gain.set(gain_val)
+
+
+def connect_device():
+    global ad7799, btn_text
+
+    if ad7799 is None:
+        ad7799 = adi.ad7799(uri="serial:" + comport.get() + ",115200")
+
+    btn_text.set("Connected!!!")
+
+
+# Create GUI
+def gui():
+
+    global dev, gain, voltage0, voltage1, voltage2, comport, ad7799, btn_text
+    global ad7799
+
+    ad7799 = None
+
+    root = tk.Tk()
+    root.title("CN0548 (Analog Devices, Inc.)")
+
+    gain = tk.StringVar()
+    voltage0 = tk.StringVar()
+    voltage1 = tk.StringVar()
+    voltage2 = tk.StringVar()
+    btn_text = tk.StringVar()
+    comport = tk.StringVar()
+
+    btn_text.set("Connect")
+
+    fr1 = tk.Frame(root)
+    fr1.pack(side=tk.LEFT, anchor="n", padx=10, pady=10)
+
+    fr2 = tk.Frame(fr1)
+    fr2.grid(row=0, column=0, pady=10)
+
+    label1 = tk.Label(fr2, text="Gain ")
+    label1.grid(row=0, column=0)
+
+    entry1 = tk.Entry(fr2, textvariable=gain)
+    entry1.grid(row=0, column=1)
+
+    label2 = tk.Label(fr2, text="Voltage0")
+    label2.grid(row=1, column=0, pady=(0, 5))
+
+    entry2 = tk.Entry(fr2, textvariable=voltage0)
+    entry2.grid(row=1, column=1)
+
+    label3 = tk.Label(fr2, text="Voltage1")
+    label3.grid(row=2, column=0)
+
+    check1 = tk.Entry(fr2, textvariable=voltage1)
+    check1.grid(row=2, column=1)
+
+    label4 = tk.Label(fr2, text="Voltage2")
+    label4.grid(row=3, column=0)
+
+    entry3 = tk.Entry(fr2, textvariable=voltage2)
+    entry3.grid(row=3, column=1)
+
+    label5 = tk.Label(fr2, text="COMPort:")
+    label5.grid(row=4, column=0)
+
+    entry4 = tk.Entry(fr2, textvariable=comport)
+    entry4.grid(row=4, column=1)
+
+    btn_sweep = tk.Button(fr2, text="Read", command=read_device)
+    btn_sweep.config(width=13, height=1, bg="orange")
+    btn_sweep.grid(row=8, column=0, pady=(10, 0))
+
+    btn_connect = tk.Button(fr2, textvariable=btn_text, command=connect_device)
+    btn_connect.config(width=13, height=1, bg="red")
+    btn_connect.grid(row=8, column=1, pady=(10, 0))
+
+    root.update_idletasks()
+
+    root.mainloop()
+
+
+# main function
+def main():
```

Comment:
Use `if __name__ == "__main__" ` style instead of `def main` then `main()`

Reply:
is this a good example to follow?
https://github.com/analogdevicesinc/pyadi-iio/blob/111d7f1448e1ae6c73b221ca1b25c59f18ecfa28/test/rf/spec.py#L165-L206

Reply:
Sure that's fine (ignore the comments though :) )